src/pykeen/checkpoints/__init__.py

```diff
@@ -0,0 +1,116 @@
+"""
+This module contains methods for deciding when to write and clear checkpoints.
+
+.. warning ::
+    While this module provides a flexible and modular way to describe a desired checkpoint behavior, it currently only
+    stores the model's weights (more precisely, its :meth:`torch.nn.Module.state_dict`).
+    Thus, it does not yet replace the full training loop checkpointing mechanism described in
+    :ref:`regular_checkpoints_how_to`.
+
+It consists of two main components: checkpoint *schedules* decide whether to write a checkpoint at a given epoch.
+If we have multiple checkpoints, we can use multiple *keep strategies* to decide which checkpoints to keep and
+which to discard. For both, we provide a set of basic rules, as well as a way to combine them via union.
+Those should be sufficient to easily model most of the desired checkpointing behaviours.
+
+Examples
+========
+
+Below you can find a few examples of how to use them inside the training pipeline.
+If you want to check before an actual training how (static) checkpoint schedules behave,
+you can take a look at :meth:`pykeen.checkpoints.final_checkpoints`
+and :meth:`pykeen.checkpoints.inspect_schedule`.
+
+Example 1
+~~~~~~~~~
+Write a checkpoint every 10 steps and keep them all.
+
+.. code-block::
+
+    from pykeen.pipeline import
```

Comment:
check the code blocks all can run

Reply:
I extracted the examples into files here: https://github.com/pykeen/pykeen/pull/1430/commits/398ff98c94b0f5bcecfaee4a19f44cf0acd63c5a

Not sure where to store them in our hierarchy - they should be outside of `src`.

Reply:
That’s cool!